examples/spark2/src/main/scala/org/apache/carbondata/examples/DataFrameSDKS3Example.scala

```diff
@@ -0,0 +1,185 @@
+    }
+    val warehouse = s"$rootPath/examples/spark2/target/warehouse"
+
+    val sparkSession = SparkSession
+      .builder()
+      .master("local")
+      .appName("SparkSessionExample")
+      .config("spark.sql.warehouse.dir", warehouse)
+      .getOrCreate()
+
+    sparkSession.sparkContext.setLogLevel("ERROR")
+
+    exampleBody(sparkSession, args)
+
+    sparkSession.stop()
+  }
+
+  def exampleBody(sparkSession: SparkSession, args: Array[String] = Array.empty): Unit = {
+
+    try {
+      val df = sparkSession.emptyDataFrame
+
+      var path = s"$rootPath/examples/spark2/target/carbon"
+      if (args.length == 1) {
+        path = args(0)
+      }
+      if (args.length == 3) {
+        path = "s3a://carbon/sdk/DFTest"
+      }
+      if (args.length > 3) {
+        path = args(3)
+      }
+
+      write(df, path, args);
+      read(df, path, args);
+    } catch {
+      case e: Exception => assert(false)
+    }
+  }
+
+  /**
+   * inherit DataFrame from other place,
+   * it need create CarbonSession for read CarbonData from S3
+   *
+   * @param df   DataFrame, including SparkConf
+   * @param path read path
+   * @param args argument, including ak, sk, endpoint
+   */
+  def read(df: DataFrame, path: String, args: Array[String]): Unit = {
+    val carbonSession = DataFrameToCarbonSession(df, path, args, 3);
+
+    val result = carbonSession
+      .read
+      .format("carbon")
+      .load(path)
+    result.show()
+    result.foreach { each =>
+      assert(each.get(0).toString.contains("city"))
+    }
+    carbonSession.stop()
+  }
+
+  /**
+   * inherit DataFrame from other place,
+   * it need create CarbonSession for write CarbonData to S3
+   *
+   * @param df   DataFrame, including SparkConf
+   * @param path write path
+   * @param args argument, including ak, sk, endpoint
+   */
+  def write(df: DataFrame, path: String, args: Array[String]): Unit = {
+    val carbonSession = DataFrameToCarbonSession(df, path, args, 4);
+
+    val rdd = carbonSession.sqlContext.sparkContext
+      .parallelize(1 to 1200, 4)
+      .map { x =>
+        ("city" + x % 8, "country" + x % 1103, "planet" + x % 10007, x.toString,
+          (x % 16).toShort, x / 2, (x << 1).toLong, x.toDouble / 13, x.toDouble / 11)
+      }.map { x =>
+      Row(x._1, x._2, x._3, x._4, x._5, x._6, x._7, x._8, x._9)
+    }
+
+    val schema = StructType(
+      Seq(
+        StructField("city", StringType, nullable = false),
+        StructField("country", StringType, nullable = false),
+        StructField("planet", StringType, nullable = false),
+        StructField("id", StringType, nullable = false),
+        StructField("m1", ShortType, nullable = false),
+        StructField("m2", IntegerType, nullable = false),
+        StructField("m3", LongType, nullable = false),
+        StructField("m4", DoubleType, nullable = false),
+        StructField("m5", DoubleType, nullable = false)
+      )
+    )
+
+    carbonSession.createDataFrame(rdd, schema)
```

Comment:
@xubo245 ,Please try rdd.toDF directly and pass columns name in selectExpr . For this you need to import carbonsession.implicits._  .